src/core/defaultappearance.js

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isName, Name } from "./primitives.js";
+import { escapePDFName } from "./core_utils.js";
+
+// Parse DA to extract font and color informations
+function parseDefaultAppearance(str) {
+  const result = {
+    fontSize: null,
+    fontName: null,
+    color: null,
+  };
+
+  if (!str) {
+    return result;
+  }
+
+  const stack = [];
+
+  for (const token of str.split(/\s+/)) {
+    if (!token) {
+      continue;
+    }
+    switch (token.charAt(0)) {
+      case "/":
+        // we've a name
+        const name = token
+          .slice(1)
+          .replace(/#([0-9a-fA-F]{2})/, (match, n) =>
+            String.fromCharCode(n.parseInt(16))
+          );
+        stack.push(Name.get(name));
+        break;
```

Comment:
Are you sure that this correctly handles all cases of malformed `Name`-data?
Note the parsing/validation, which is another argument for re-using that functionality here, in e.g. https://github.com/mozilla/pdf.js/blob/35845d1bbb7d6b581e48ddd8d8f1a96ab9a931d2/src/core/parser.js#L1075-L1120